server/command.go

```diff
@@ -79,40 +122,160 @@ func (p *Plugin) validateCommand(action string, parameters []string) string {
 		if len(parameters) > 0 {
 			return "`delete_channel_welcome` command does not accept any extra parameters"
 		}
+	case commandTriggerSetTeamWelcome:
+		if len(parameters) == 0 {
+			return "`" + commandTriggerSetTeamWelcome + "` command requires the message to be provided"
+		}
+	case commandTriggerGetTeamWelcome:
+		if len(parameters) > 0 {
+			return "`" + commandTriggerGetTeamWelcome + "` command does not accept any extra parameters"
+		}
+	case commandTriggerDeleteTeamWelcome:
+		if len(parameters) > 0 {
+			return "`" + commandTriggerDeleteTeamWelcome + "` command does not accept any extra parameters"
+		}
 	}
 
 	return ""
 }
 
-func (p *Plugin) executeCommandPreview(teamName string, args *model.CommandArgs) {
-	found := false
-	for _, message := range p.getWelcomeMessages() {
-		if message.TeamName == teamName {
-			if err := p.previewWelcomeMessage(teamName, args, *message); err != nil {
-				p.postCommandResponse(args, "error occurred while processing greeting for team `%s`: `%s`", teamName, err)
-				return
-			}
+func (p *Plugin) validatePreviewPrivileges(teamID string, args *model.CommandArgs) bool {
+	_, teamMemberErr := p.API.GetTeamMember(teamID, args.UserId)
+	if teamMemberErr != nil {
+		if teamMemberErr.StatusCode == 404 {
+			p.postCommandResponse(args, "You are not a member of that team.")
+			return false
+		}
+		p.postCommandResponse(args, "error occurred while getting the Team Admin Role `%s`: `%s`", teamID, teamMemberErr)
+		return false
+	}
+	doesUserHavePrivileges := p.checkForProperPrivileges(args, args.UserId, teamID)
+
+	return doesUserHavePrivileges
+}
+
+func (p *Plugin) checkForProperPrivileges(args *model.CommandArgs, userID string, teamID string) bool {
+	isSysadmin, sysAdminError := p.hasSysadminRole(userID)
+	isTeamAdmin, teamAdminError := p.hasTeamAdminRole(userID, teamID)
+
+	if sysAdminError != nil {
+		p.postCommandResponse(args, "error occurred while getting the System Admin Role `%s`: `%s`", teamID, sysAdminError)
+		return false
+	}
+	if teamAdminError != nil {
+		p.postCommandResponse(args, "error occurred while getting the Team Admin Role `%s`: `%s`", teamID, teamAdminError)
+		return false
+	}
+	if !isSysadmin && !isTeamAdmin {
+		p.postCommandResponse(args, "You do not have the proper privileges to control this Team's welcome messages.")
+		return false
+	}
+	return true
+}
+
+func (p *Plugin) showDynamicMessages(args *model.CommandArgs) bool {
```

Comment:
Returning a `bool` make the code hard to read. Should it return a `(string, bool)` instead?